packages/shared/test-suites/types.ts

```diff
@@ -0,0 +1,301 @@
+
+  // Actions that were part of this test, grouped by section (or "hook")
+  events: Record<TestSectionName, TestEvent[]>;
+
+  // Note that the client does not necessarily have any expectations of ordering here;
+  // we will likely order results lexicographically, by title.
+  result: TestResult;
+
+  source: {
+    // Example test:
+    //   describe("outer", () => {
+    //     describe("inner", () => {
+    //       it("test name", () => {
+    //         // Test code here...
+    //       });
+    //     });
+    //   });
+
+    // If a test is inside of one or more describe() blocks
+    // For the example test above, this would be ["outer", "inner"]
+    scope: string[];
+
+    // For Cypress, the string passed to the it() block
+    // For Playwright, the string passed to the test() block
+    // For the example test above, this would be "test name"
+    title: string;
+  };
+
+  // Defines the precise boundaries of the test run (including beforeEach and afterEach blocks);
+  // this value comes from annotation data
+  timeStampedPointRange: TimeStampedPointRange;
+}
+
+export interface UserActionEvent {
+  data: {
+    category: "assertion" | "command" | "other";
+
+    command: {
+      arguments: string[];
+      name: string;
+    };
+
+    // An error that occurred while executing this action (if any)
+    error: TestError | null;
+
+    // Used to associate related annotations
+    id: string;
+
+    // Used to associate chained commands
+    parentId: string | null;
+
+    result: {
+      timeStampedPoint: TimeStampedPoint;
+      variable: string;
+    } | null;
+
+    // If an action is somewhere other than the main test body;
+    // for example, before/after actions have different scopes
+    scope: string[] | null;
+
+    viewSourceTimeStampedPoint: TimeStampedPoint;
+  };
+
+  // Precisely defines the start/stop execution points (and times) for the action
+  timeStampedPointRange: TimeStampedPointRange;
+
+  type: "user-action";
+}
+
+export interface NavigationEvent {
+  // Data needed to render this event
+  data: {
+    url: string;
+  };
+
+  // The precise time this event occurred; not that events have no duration
+  timeStampedPoint: TimeStampedPoint;
+
+  type: "navigation";
+}
+
+export interface NetworkRequestEvent {
+  // Data needed to render this event
+  data: {
+    request: {
+      method: string;
+      url: string;
+    };
+    response: {
+      status: number;
+    } | null;
+  };
+
+  // The precise time this event occurred; not that events have no duration
+  timeStampedPoint: TimeStampedPoint;
+
+  type: "network-request";
+}
+
+export type TestEvent = UserActionEvent | NavigationEvent | NetworkRequestEvent;
+
+export type TestError = {
+  name: string;
+  message: string;
+};
+
+// Intermediate types the frontend supports temporarily (until backend data has finished migrating)
+
+export type IncrementalTestRecording = Omit<TestRecording, "events" | "timeStampedPointRange"> & {
```

Comment:
:question: what makes this "Incremental"?  Is it that it's a type meant to be replaced later?

("Incremental" probably isn't the first name that would come to my mind, but not sure I immediately have a different suggestion. "Placeholder"? "Temporary"?)

Reply:
This is not the final data schema I proposed. It's all that we had time to build before June 12th. So "incremental" indicates that it's...an in between step.